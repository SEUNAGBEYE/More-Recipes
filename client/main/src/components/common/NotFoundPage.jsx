import React from 'react';
import { Link } from 'react-router-dom';
import Exclamation from '../recipes/Exclamation';

const NotFoundPage = () => (
  <div>
    <Exclamation>
      <p className="text-muted" style={{ marginTop: 200, fontSize: 20 }}> 404 Sorry Page Not Found <Link to="/">Go Home</Link></p>
    </Exclamation>
  </div>
);
export default NotFoundPage;

