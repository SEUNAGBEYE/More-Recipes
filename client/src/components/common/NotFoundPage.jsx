import React from 'react';
import { Link } from 'react-router-dom';
import Navbar from '../common/Navbar';
import Footer from '../common/Footer';
import Exclamation from '../recipes/Exclamation';

const NotFoundPage = () => (
  <div>
    <Navbar />
    <Exclamation>
      <p className="text-muted" style={{ marginTop: 200, fontSize: 20 }}> 404 Sorry Page Not Found <Link to="/">Go Home</Link></p>
    </Exclamation>
    <Footer />
  </div>
);
export default NotFoundPage;

