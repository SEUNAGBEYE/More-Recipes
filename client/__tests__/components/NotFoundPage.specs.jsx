import React from 'react';
import { shallow, mount } from 'enzyme';
import { Link } from 'react-router';
import NotFoundPage from '../../main/src/components/common/NotFoundPage';

describe('# NotFoundPage', () => {
  const wrapper = shallow(<NotFoundPage />);
  it('should render successfully', (done) => {
    expect(wrapper).toBeDefined();
    const paragraph = wrapper.find('p');
    expect(paragraph.length).toEqual(1)
    expect(wrapper).toMatchSnapshot();
    done();
  });
});
