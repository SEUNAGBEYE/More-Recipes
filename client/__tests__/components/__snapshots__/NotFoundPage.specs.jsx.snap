// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# NotFoundPage should render successfully 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NotFoundPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Exclamation>
        <p
          className="text-muted"
          style={
            Object {
              "fontSize": 20,
              "marginTop": 200,
            }
          }
        >
          404 Sorry Page Not Found 
          <Link
            replace={false}
            to="/"
          >
            Go Home
          </Link>
        </p>
      </Exclamation>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p
          className="text-muted"
          style={
            Object {
              "fontSize": 20,
              "marginTop": 200,
            }
          }
        >
          404 Sorry Page Not Found 
          <Link
            replace={false}
            to="/"
          >
            Go Home
          </Link>
        </p>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "404 Sorry Page Not Found ",
            <Link
              replace={false}
              to="/"
            >
              Go Home
            </Link>,
          ],
          "className": "text-muted",
          "style": Object {
            "fontSize": 20,
            "marginTop": 200,
          },
        },
        "ref": null,
        "rendered": Array [
          "404 Sorry Page Not Found ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Go Home",
              "replace": false,
              "to": "/",
            },
            "ref": null,
            "rendered": "Go Home",
            "type": [Function],
          },
        ],
        "type": "p",
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Exclamation>
          <p
            className="text-muted"
            style={
              Object {
                "fontSize": 20,
                "marginTop": 200,
              }
            }
          >
            404 Sorry Page Not Found 
            <Link
              replace={false}
              to="/"
            >
              Go Home
            </Link>
          </p>
        </Exclamation>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <p
            className="text-muted"
            style={
              Object {
                "fontSize": 20,
                "marginTop": 200,
              }
            }
          >
            404 Sorry Page Not Found 
            <Link
              replace={false}
              to="/"
            >
              Go Home
            </Link>
          </p>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "404 Sorry Page Not Found ",
              <Link
                replace={false}
                to="/"
              >
                Go Home
              </Link>,
            ],
            "className": "text-muted",
            "style": Object {
              "fontSize": 20,
              "marginTop": 200,
            },
          },
          "ref": null,
          "rendered": Array [
            "404 Sorry Page Not Found ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Go Home",
                "replace": false,
                "to": "/",
              },
              "ref": null,
              "rendered": "Go Home",
              "type": [Function],
            },
          ],
          "type": "p",
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
